compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/destructure-array-assignment-to-context-var.expect.md

```diff
@@ -30,15 +30,15 @@ import { identity } from "shared-runtime";
 function Component(props) {
   const $ = _c(4);
   let x;
-  if ($[0] !== props.value) {
+  if ($[0] !== props) {
```

Comment:
Hmmm why does this regress? `props.value` is unconditionally accessed both at the top-level and in the function expression so i'd expect that to be the inferred dep

Reply:
This is the same case as https://github.com/facebook/react/pull/30894#discussion_r1747781289, will change to return jsx. Happy to also improve `compilationMode: 'all'` in a followup

Reply:
got it, sounds good!